lib/model/message.dart

```diff
@@ -302,4 +305,24 @@ class MessageStoreImpl with MessageStore {
       view.notifyListenersIfMessagePresent(event.messageId);
     }
   }
+
+  void handleSubmessageEvent(SubmessageEvent event) {
+    final poll = messages[event.messageId]?.poll;
+    if (poll == null) return;
+    PollEventSubmessage? pollEventSubmessage;
+    try {
+      pollEventSubmessage = PollEventSubmessage.fromJson(jsonDecode(event.content) as Map<String, Object?>);
+    } on TypeError catch (e) {
+      assert(debugLog('Malformed submessage event data for poll: $e\n${jsonEncode(event)}')); // TODO(log)
+      return;
+    }
```

Comment:
Let's move these details to live on `Poll` — that seems like the natural home for them.

Then the signature of that method can be `void handleSubmessageEvent(SubmessageEvent event)`, just like its caller here, and following our usual convention for methods that handle events.